python/pyspark/sql/tests/test_pandas_udf_scalar.py

```diff
@@ -850,6 +850,36 @@ def test_close(batch_iter):
             with self.assertRaisesRegexp(Exception, "reached finally block"):
                 self.spark.range(1).select(test_close(col("id"))).collect()
 
+    def test_scalar_iter_udf_close_early(self):
+        tmp_dir = tempfile.mkdtemp()
+
+        @pandas_udf('int', PandasUDFType.SCALAR_ITER)
+        def test_close(batch_iter):
+            generator_exit_caught = False
+            try:
+                for batch in batch_iter:
+                    yield batch
+                    time.sleep(0.5)
```

Comment:
Do not put a plain sleep here. Sleep after all just don't let the generator exit.

Reply:
Sleep at last do not work. The sleep cmd cannot be interrupt (during sleep GeneratorExit cannot be raised)